src/test/java/Homework24Test.java

```diff
@@ -0,0 +1,85 @@
+import org.openqa.selenium.remote.RemoteWebDriver;
+import org.openqa.selenium.support.ui.FluentWait;
+import org.openqa.selenium.support.ui.Wait;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.Assert;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.Parameters;
+import org.testng.annotations.Test;
+import page.HomePage;
+import pageFactory.LoginPageFromFactory;
+
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.time.Duration;
+
+public class Homework24Test {
+    public static WebDriver driver = null;
+    public ChromeOptions options = new ChromeOptions();
+    public WebDriverWait wait;
+    public Wait<WebDriver> fluentWait;
+    public Actions actions = null;
+
+    @BeforeSuite
+    static void setupClass() {
+        //WebDriverManager.chromedriver().setup();
+        WebDriverManager.firefoxdriver().setup();
+    }
+
+    @BeforeMethod
+    @Parameters({"BaseURL"})
+    public void launchBrowser(String baseURL) throws MalformedURLException {
+        driver = BaseTest.pickBrowser(System.getProperty("browser"));
+        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
+        driver.manage().window().maximize();
+        wait = new WebDriverWait(driver, Duration.ofSeconds(10));
+        actions = new Actions(driver);
+        fluentWait = new FluentWait<WebDriver>(Homework24Test.driver)
+                .withTimeout(Duration.ofSeconds(5))
+                .pollingEvery(Duration.ofMillis(200));
+        navigateToPage(baseURL);
+    }
+
+
+    public void navigateToPage(String url) {
+        driver.get(url);
+    }
+
+    public static WebDriver pickBrowser(String browser) throws MalformedURLException {
+        DesiredCapabilities caps = new DesiredCapabilities();
+        String gridURL = "http://192.168.1.108:4444";
+
+        switch (browser) {
+            case "firefox":
+                WebDriverManager.firefoxdriver().setup();
+                return driver = new FirefoxDriver();
+            case "grid-chrome":
+                caps.setCapability("browserName", "chrome");
+                return driver = new RemoteWebDriver(URI.create(gridURL).toURL(), caps);
+            default:
+                WebDriverManager.chromedriver().setup();
+                ChromeOptions chromeOptions = new ChromeOptions();
+                chromeOptions.addArguments("--remote-allow-origins=*");
+                return driver = new ChromeDriver(chromeOptions);
+        }
+    }
+    @Test
+    public void loginValidEmailPassword()  {
+        LoginPageFromFactory loginPageFromFactory = new LoginPageFromFactory(driver);
+        HomePage homePage = new HomePage(driver);
+        loginPageFromFactory.provideEmail("kristina.sarkisyan@testpro.io").providePassword("o8URUDnW").clickSubmit();
+        Assert.assertTrue(homePage.getUserAvatar().isDisplayed());
+
+
```

Comment:
**Test in Page Object Model (POM)**

![Clean_Test](https://user-images.githubusercontent.com/104761482/236863813-f30d7bfc-e89d-4419-bdea-4b1517155e39.png)

![Screenshot_5](https://github.com/KristinaSarkisyan/A59-QA-Automation-Framework/assets/104761482/7de350ef-4072-4120-9ead-eafcb4cc9cb6)